src/AudioManager.test.ts

```diff
@@ -0,0 +1,182 @@
+      duckVolumeLevel: 0.2,
+      normalVolumeLevel: 1,
+      sttService: "browser",
+      streamTimeslice: 100,
+    };
+
+    // Create a new instance of AudioManager
+    audioManager = new AudioManager(mockOptions);
+
+    // Retrieve the instance of MediaAudio
+    // eslint-disable-next-line prefer-destructuring, @typescript-eslint/no-unsafe-assignment
+    mockMediaAudioInstance = (MediaAudio as jest.Mock).mock.instances[0];
+    mockMediaAudioInstance.isPlaying = true;
+  });
+
+  // Audio Input Service Tests
+  test("inputServiceStartListening should call startListening and duck volume if media is playing", () => {
+    audioManager.mediaAudio.isPlaying = true;
+    audioManager.inputServiceStartListening();
+    expect(audioManager.audioInputsService.startListening).toHaveBeenCalled();
+    expect(audioManager.mediaAudio.volume).toBe(mockOptions.duckVolumeLevel);
+  });
+
+  test("inputServiceStartListening should call startListening and not change volume if media is not playing", () => {
+    audioManager.mediaAudio.isPlaying = false;
+    audioManager.inputServiceStartListening();
+    expect(audioManager.audioInputsService.startListening).toHaveBeenCalled();
+    expect(audioManager.mediaAudio.volume).not.toBe(
+      mockOptions.duckVolumeLevel,
+    );
+  });
+
+  test("inputServiceStopListening should call stopListening and set volume to normal if media is playing", () => {
+    audioManager.mediaAudio.isPlaying = true;
+    audioManager.inputServiceStopListening();
+    expect(audioManager.audioInputsService.stopListening).toHaveBeenCalled();
+    expect(audioManager.mediaAudio.volume).toBe(mockOptions.normalVolumeLevel);
+  });
+
+  test("inputServiceStopListening should call stopListening and not change volume if media is not playing", () => {
+    audioManager.mediaAudio.isPlaying = false;
+    audioManager.inputServiceStopListening();
+    expect(audioManager.audioInputsService.stopListening).toHaveBeenCalled();
+    expect(audioManager.mediaAudio.volume).toBe(mockOptions.normalVolumeLevel);
+  });
+
+  test("inputServiceConnect should call connect with the provided token", () => {
+    const token = "testToken";
+    audioManager.inputServiceConnect(token);
+    expect(audioManager.audioInputsService.connect).toHaveBeenCalledWith(token);
+  });
+
+  test("inputServiceResetTimeout should call resetTimeout with the provided timeout", () => {
+    const timeout = 5000;
+    audioManager.inputServiceResetTimeout(timeout);
+    expect(audioManager.audioInputsService.resetTimeout).toHaveBeenCalledWith(
+      timeout,
+    );
+  });
+
+  // Browser STT Service Tests
+  test("browserIsSupported should return isSupported from BrowserSttService", () => {
+    expect(audioManager.browserIsSupported()).toBe(
+      audioManager.audioInputsBrowser.isSupported,
+    );
+  });
+
+  test("browserStartListening should call startListening and duck volume if media is playing", () => {
+    audioManager.mediaAudio.isPlaying = true;
```

Comment:
If you are going to test changes to mediaAudio.volume I'd expect to see an original value in the test setup, a value different to the one expected after the "act"